samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/print/PDFAssessmentBean.java

```diff
@@ -642,6 +644,8 @@ else if (item.getItemData().getTypeId().equals(TypeIfc.ESSAY_QUESTION)) {
 					contentBuffer.append(item.getKey());
 				else
 					contentBuffer.append("--------");
+			} else if(item.getItemData().getTypeId().equals(TypeIfc.CALCULATED_QUESTION)){
```

Comment:
I realize that all the existing code is like this, but if item or getItemData or getTypeId can be/return null, this statement will generate an NPE.

Best practise is to do the opposite, compare the constant to the value, because the constant will always have a value:

TypeIfc.CALCULATED_QUESTION.equals( item.getItemData().getTypeId() )